adapter/test_adapter.py

```diff
@@ -13,13 +18,46 @@ def mock_slack_adapter(self):
         with patch('slack_bolt.App') as MockApp:
             mock_app = MockApp()
             mock_chatbot = MagicMock(spec=ChatOpenAI)
-            slack_adapter = SlackAdapter(mock_app, mock_chatbot)
-            return mock_app, mock_chatbot, slack_adapter
+            mock_bot_controller = MagicMock(spec=BotController)
+            slack_adapter = SlackAdapter(mock_app, mock_chatbot, mock_bot_controller)
+            return mock_app, mock_chatbot, mock_bot_controller, slack_adapter
```

Comment:
This might be better refactored into separate functions eventually since returning a 4-tuple is quite odd under most circumstances. Most of the test functions only utilize 1-2 dependencies here anyway